Products/ZenModel/migrate/addDeviceIpAddrRelation.py

```diff
@@ -40,6 +36,22 @@ def cutover(self, dmd):
             except (Exception,):
                 continue
 
+        log.info("Updating relationships on Devices")
+        for dev in dmd.Devices.getSubDevices():
+            try:
+                dev.buildRelations()
+                if dev.manageIp:
+                    ipobj = dev.getNetworkRoot().findIp(dev.manageIp)
```

Comment:
perhaps we should check to see if the device already has an ipAddress object in it's relation?

Reply:
I suppose there should be no ipAddress object in device relation before running this migration.

Reply:
That assumes this migration will only ever run once.  Probably likely, but in the case where the migration runs but the upgrade fails after, is it possible that the migration will re-run?  We don't really want to end up in a situation where it tries to make a bunch of duplicate IpAddress objects.

Reply:
There's a `if ipobj` statement below which sets up the relation if the IpAddress object already exists.